src/ibmca_pkey.c

```diff
@@ -155,7 +155,7 @@ static int ibmca_x25519_keygen(EVP_PKEY_CTX *c, EVP_PKEY *pkey)
         free(private);
     }
     if (ctx != NULL)
-        ica_x25519_ctx_del(&ctx);
+        p_ica_x25519_ctx_del(&ctx);
```

Comment:
The check for ctx not null is old code, but just to mention: the ica_x25519_ctx_del does this check internally.

Reply:
There are a couple more places below ....

Reply:
Okay, I only changed from linking directly against libica to the function pointer call.  Did not check the need for the `if`, but I doubt it will have a big impact.  Did libica always check for `NULL` in `ica_x25519_ctx_del`?

Reply:
Yes, the check was there since the functions were introduced. But as always, the impact is rather small. You may leave it as it is, I just noticed the fact.